x-pack/solutions/security/plugins/security_solution/public/data_view_manager/hooks/use_data_view.ts

```diff
@@ -20,7 +20,7 @@ import type { SharedDataViewSelectionState } from '../redux/types';
  * selected data view.
  */
 export const useDataView = (
```

Comment:
May it happend `dataViewId` returned by `sourcererAdapterSelector()` matches a removed Data View?

In that case `await dataViews?.get(dataViewId)` would silently fail since it's inside a `useEffect()`.

Reply:
Yea, I agree with @lgestc, but if we want to be defensive, we could do the following:

```
try {
  const currDv = await dataViews?.get(dataViewId);
   setRetrievedDataView(currDv);
  } catch (error) {
  // show notiification
  setRetrievedDataView(...undefined or default security dv)
  }
```

Reply:
we have a toast now